docs/_includes/Statements.md

```diff
@@ -239,6 +240,305 @@ Note that the form
 
 is diagnosed as a label in which the user forgot the **label** keyword.
 
+## Update with Failure Statement (`:-`)
+````grammar
+UpdateFailureStmt  =
+    [ Lhs { "," Lhs } ]
+    ":-"
+    [ "expect" ]
+    Expression(allowLemma: false, allowLambda: false) { "," Rhs } 
+````
+
+A `:-` statement is similar to a `:=` statement, but allows for immediate return if a failure is detected.
+This is a language feature somewhat analogous to exceptions in other languages.
+
+An update-with-failure statement uses _failure-compatible_ types.
+A failure-compatible type is a type that has the following members (each with no in-parameters and one out-parameter):
+
+ * a function method `IsFailure()` that returns a `bool`
+ * a function method `PropagateFailure()` that returns a value assignable to the first out-parameter of the caller
+ * an optional method or function `Extract()`
+
+A failure-compatible type with an `Extract` method is called _value-carrying_.
```

Comment:
`Extract` can be either a method or a function, right? Same comments applies to every mention of `Extract`.

Reply:
Changed to member everywhere